### PR TITLE
Update NewModuleScript

### PR DESCRIPTION
**Pull request by shubham7109**
## Description
PR to update the `NewModuleScript.jar` to create new sample library modules for the sample viewer project. 

##### Note: 
- This PR also fixes the issue of empty folders being considered as sample modules. 

## Links and Data
Sample Issue: [#4538](https://devtopia.esri.com/runtime/kotlin/issues/4538)

## How to Test:

Run the jar file using the following steps: 
```console
$ java -jar tools/NewModuleScript.jar
Enter the name of the sample with spaces (Eg. "Display new map"):    
....
```
- Run Gradle Sync
- Run Sample Viewer project

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -66,7 +66,9 @@ android {
 dependencies {
     for (sampleFile in file("../samples").listFiles()!!) {
         if (sampleFile.isDirectory) {
-            implementation(project(":samples:" + sampleFile.name))
+            if (sampleFile.listFiles()?.find { it.name.equals("build.gradle.kts") } != null){
```
Comment (TADraeseke) [inline app/build.gradle.kts:69]: Maybe reads better with a contains check like:

`if (sampleFile.isDirectory && sampleFile.listFiles()?.contains(File("build.gradle.kts")) == true) {`

File: settings.gradle.kts [hunk 2]
```diff
@@ -32,7 +32,9 @@ include(":samples-lib")
 
 // dynamically include all sample libraries
 File("samples/").listFiles()?.filter { !it.name.contains(".DS_Store") }?.forEach { sampleFolder ->
-    // include all of the samples, which have been changed into libs of the sample viewer app
-    include(":samples:" + sampleFolder.name)
-    project(":samples:" + sampleFolder.name).projectDir = file("samples/" + sampleFolder.name)
+    if (sampleFolder.listFiles()?.find { it.name.equals("build.gradle.kts") } != null){
```
Comment (TADraeseke) [inline settings.gradle.kts:35]: as above, could use contains

**Review by TADraeseke — APPROVED** [review]
Works great--what an improvement! Just one small suggestion below

File: tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                        "src/main/java/com/esri/arcgismaps/sample/$sampleWithoutSpaces/${sampleNameCamelCase}Screen.kt",
+                        "src/main/java/com/esri/arcgismaps/sample/$sampleWithoutSpaces/MainActivity.kt"
+                    ],
+                    "title": "$sampleName"
+                }
+                
+                """.trimIndent(), StandardCharsets.UTF_8
+        )
+    } catch (e: IOException) {
+        e.printStackTrace()
```
Comment (sorenoid) [inline tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt:233]: This will print the stacktrace twice, it probably isn't necessary.

Comment (sorenoid) [reply tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt:233]: same elsewhere...

File: tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+        exitProgram(e)
+    }
+
+    //Update MainActivity.kt
+    file =
+        File("$samplesRepoPath/samples/$sampleWithHyphen/src/main/java/com/esri/arcgismaps/sample/$sampleWithoutSpaces/MainActivity.kt")
+    try {
+        var fileContent = FileUtils.readFileToString(file, StandardCharsets.UTF_8)
+        fileContent = fileContent.replace("Copyright 2023", "Copyright " + Calendar.getInstance()[Calendar.YEAR])
```
Comment (sorenoid) [inline tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt:267]:
```suggestion
        fileContent = fileContent.replace("Copyright 2024", "Copyright " + Calendar.getInstance()[Calendar.YEAR])
```

File: tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+        e.printStackTrace()
+        exitProgram(e)
+    }
+
+    //Update MainScreen.kt
+    file =
+        File("$samplesRepoPath/samples/$sampleWithHyphen/src/main/java/com/esri/arcgismaps/sample/$sampleWithoutSpaces/screens/${sampleNameCamelCase}Screen.kt")
+    try {
+        var fileContent = FileUtils.readFileToString(file, StandardCharsets.UTF_8)
+        fileContent = fileContent.replace("Copyright 2023", "Copyright " + Calendar.getInstance()[Calendar.YEAR])
```
Comment (sorenoid) [inline tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt:298]:
```suggestion
        fileContent = fileContent.replace("Copyright 2024", "Copyright " + Calendar.getInstance()[Calendar.YEAR])
```

File: tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+}
+
+/**
+ * Exits the program with error -1 if it encounters an error
+ * @param e Error message to display
+ */
+private fun exitProgram(e: Exception) {
+    println("Error creating the sample: " + e.message)
+    println("StackTrace:")
+    e.printStackTrace()
```
Comment (sorenoid) [inline tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt:185]: You might consider not printing the stacktrace, or passing a default boolean parameter to print it. It seems like most usage of this function is preceded by printing the stacktrace.

File: tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt (hunk 4 above)
Comment (shubham7109) [reply tools/NewModuleScript/src/main/kotlin/NewModuleScript.kt:267]: The template uses `2023`, this line is to replace that content with the _"current"_ year. (2024/2025/...)

**Comment by shubham7109** [comment]
@sorenoid Thanks for pointing out the overuse of try/catches in the script. I have updated the code to contain one try/catch and one printStackTrace enclosed within the `exitProgram` function.

**Comment by shubham7109** [comment]
Tested the changes locally, merging this into `v.next`. Thanks @sorenoid @TADraeseke!

**Pull request merged**